### PR TITLE
Add `manual-strip` lint

### PR DESCRIPTION
**Pull request by mikerite**
Add `manual-strip` lint.

changelog: Add `manual-strip` lint

### CODE REVIEW

File: clippy_lints/src/manual_strip.rs [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+    if_chain! {
+        if let ExprKind::MethodCall(_, _, [arg], _) = expr.kind;
+        if let Some(method_def_id) = cx.typeck_results().type_dependent_def_id(expr.hir_id);
+        if match_def_path(cx, method_def_id, &paths::STR_LEN);
+        then {
+            Some(arg)
+        }
+        else {
+            None
+        }
```
Comment (matthiaskrgr) [inline clippy_lints/src/manual_strip.rs:130]: nit: formatting

**Comment by flip1995** [comment]
Testing rustbot

@rustbot modify labels: +S-waiting-on-author -S-waiting-on-review

**Comment by matthiaskrgr** [comment]
thanks!
@bors r+

**Comment by jhpratt** [comment]
This is quite broad, as it doesn't just test for the simple case. The lint is triggering in a block of code that I have where not only do I need the access the stripped part, but there's also side effects.

**Pull request merged**